.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+repos:
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.9.2
+    hooks:
+      - id: flake8
+        args: [--config=setup.cfg]
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: no-commit-to-branch
```

Comment:
This is mostly for anyone with push access to not accidentally commit to `main` and push to GH. `main` and `master` are forbidden [by default](https://github.com/pre-commit/pre-commit-hooks#no-commit-to-branch) so no further configuration is needed.

Reply:
Sometimes, people from the infra team or the OSS team need to push stuff directly to `main` for urgent fixes, e.g. https://github.com/pytorch/vision/pull/3711#issuecomment-825823893
I doubt they will have the pre-commit hooks installed, but this might get in their way.

Also, `main` is a protected branch so even regular maintainers like us can't push to it directly anyway. So maybe we can remove this one?

Reply:
> Also, `main` is a protected branch so even regular maintainers like us can't push to it directly anyway.

Are you sure about that? So if I would do

```bash
git checkout main
git commit --allow-empty -m "test commit"
git push
```

I would get rejected? If yes, there is no point in having this. I'm not going to try this though :stuck_out_tongue: Let's remove it for now and if we ever such a case revisit this.

Reply:
I'm fairly certain, but like you, I didn't want to try and make sure ^^

The GUI is pretty clear though, if you try to edit any file from GitHub:

![image](https://user-images.githubusercontent.com/1190450/132841807-80a5f56d-dfc6-45bd-919f-e3aac502c4c6.png)